packages/adapter-components/src/elements/soap/type_elements/types_generator.ts

```
@@ -0,0 +1,80 @@
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { ObjectType } from '@salto-io/adapter-api'
+import { logger } from '@salto-io/logging'
+import { values } from '@salto-io/lowerdash'
+import _ from 'lodash'
+import * as soap from 'soap'
+import { RestrictionElement, SchemaElement } from 'soap/lib/wsdl/elements'
+import { convertComplexTypes } from './complex_types_converter'
+import { linkTypes } from './object_types_linker'
+import { convertToNamespaceName } from './utils'
+
+const log = logger(module)
+
+/**
+ * This methods treats simpleType as aliases to another types
+ * (since simpleType can only add restrictions and not elements or attributes)
+ */
+const getTypeAliases = (schema: SchemaElement): Record<string, string> =>
+  _(schema.types)
+    .entries()
+    .map(([name, simpleType]) => {
+      const element = simpleType.children[0]
+      if (element instanceof RestrictionElement) {
+        return [`${schema.$targetNamespace}:${name}`, convertToNamespaceName(element.$base, element.schemaXmlns ?? {}, schema.$targetNamespace)]
+      }
+      log.warn(`Received unexpected simple type element restriction: ${element?.name}`)
+      return undefined
+    })
+    .filter(values.isDefined)
+    .fromPairs()
+    .value()
+
+
+/**
+ * Converts WSDL to object types.
+ *
+ * This method knowingly does not support:
+ *  - complex type inside complex type.
+ *  - elements with 'ref' attribute.
+ * Also, currently if two there are two complex types with the same name in different namespace,
+ * two object types with the same elem id will be created.
```

we have seen in the past cases where different namespaces had types with the same name. so this is a pretty important limitation. we don't have to make it work here, but maybe we should throw an explicit error when that happens?